### ISSUE TITLE
Screen flickering inside of the window

### ISSUE THREAD

**Issue by RecNes** [dbeaver/dbeaver#4859]
- Bug report

Screen is flickering while running dbeaver. Video link below is more descriptive

https://streamable.com/1kg7a

**Comment by tati-kru** [comment]
@RecNes could you give us a bit more details: 
- OS
- DBeaver version
- Database name
- Specific details (connection type, additional extensions, specific Java version, etc)
- Steps to reproduce if exist

Thank you

**Comment by RecNes** [comment]
Flickering is not occurring at specific window. It's annoyingly happening in any section. This was happen when Open Java environment were available. Than I'd switched from Open Java to Oracle's Java, but there has been no change. 

**OS:** Linux Debian 9 Buster
**Kernel:** 4.18.0-3-amd64 #1 SMP Debian 4.18.20-2 (2018-11-23) x86_64 GNU/Linux
**DBeaver:** 5.3.1.201812232236 (Version is not relevant)
**Java:** java 11.0.1 2018-10-16 LTS
Java(TM) SE Runtime Environment 18.9 (build 11.0.1+13-LTS)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11.0.1+13-LTS, mixed mode)
**DatabaseName:** Not relevant
**DB Drivers:** Not relevant
**Connection Type:** Not relevant
**Extentions:** No additional extensions installed
""Xorg Info:"" Detailed X information has attached.

**Steps To Reproduce:** Just open dbeaver and begin to use.

[Xorg.0.log](https://github.com/dbeaver/dbeaver/files/2709604/Xorg.0.log)

**Comment by serge-rider** [comment]
This seems to be this bug: https://bugs.eclipse.org/bugs/show_bug.cgi?id=517671

For now the only proposed workaround on Debian 9  is to use Java 8 (perhaps Java 9 or 10 will also work).

**Comment by MedShake** [comment]
Here the same problem with Linux Mint Cinnamon 19 (all updates done) and DBeaver 5.3.2.

B.

**Comment by serge-rider** [comment]
Please check for workaround in #4991

**Comment by RecNes** [comment]
> Please check for workaround in #4991

Still little bit flickering but on insignificant level. The workaround is work for me.

**Comment by MedShake** [comment]
Ok, not realy easy to understand, but I pasted
`export GTK_IM_MODULE=ibus`
in ~/.profile and It seems to work !
Thanks !

**Comment by felangga** [comment]
ibus mode reducing the flicker, but still happens to me sometimes

**Comment by locojuhi** [comment]
It's weird... At my office works perfectly fine with xubuntu but In home with linux mint start "blink"... I don't know how exactly describe it...

**Comment by RecNes** [comment]
> > Please check for workaround in #4991
> 
> Still little bit flickering but on insignificant level. The workaround is work for me.

The workaround has some strange side affect on my keyboard layout. Because of I'm using Turkish keyboard. When I set the IBus workaround, some of characters are start to work strange.
Such as; if I try to write "kasım" while IBus is set, it starts well, I can write straight "kas". but when I hit the "ı" button cursor is jump to start of the word and put "ı" to very begining of the word and jumps back to end of the word and continue to write other character(s).

Finally I got "ıkasm" instead of "kasım" on screen.

This is happening only on "ışğüöçİ" characters. Others are working fine. 

That's why I roll back from IBus mode to normal. So flickering is still going on.
If any one have a solution for that strange keyboard problem I can switch back to IBus mode.

**Comment by AxelEriksson0** [comment]
I too, have screen flickering:

- Linux Mint 19.1 Cinnamon
- DBeaver 5.3.5
- Connected to MySQL 5.7

**Comment by luisAzcuaga** [comment]
I Pasted the ibus on my ~/.profile file and the flicking persist.

 luis@Newton
OS: Debian 9.8 stretch
Kernel: x86_64 Linux 4.9.0-8-amd64
Uptime: 6m
Shell: zsh 5.3.1
Resolution: 1366x768
DE: Xfce
WM: Xfwm4
GTK Theme: Adwaita-dark [GTK2]
CPU: Intel Core i3-5005U @ 4x 1.9GHz [53.0°C]
GPU: Mesa DRI Intel(R) HD Graphics 5500 (Broadwell GT2) 
RAM: 1144MiB / 5881MiB


I'm currently on DBeaver 6 (Even on DBeaver 5 the problem existed but was way less anoying). As soon as a I click Data inside any Table, the data dissappears.

**Comment by g3n1k** [comment]
disable iBus and restart DBeaver

**Comment by StephanoGeorge** [comment]
Starting a month ago, I also encountered this problem.
OS: Arch Linux x86_64
DBeaver: ee, version 7.1.0
Java: openjdk version "1.8.0_262"
DE: Plasma5
WM: KWin
Input: Fcitx5
`export GTK_IM_MODULE=ibus` and start `dbeaver-ee` in terminal did not work.

**Comment by lavalade** [comment]
I have had the same bug.  
To make BÉPO keyboard layout works properly, I switched from `ibus` to `xim`, but this have made DBeaver tabs blinking (especially SQL scripts).  
To fix this, I needed to do two things (need to restart the computer afterward):
1. `export GTK_IM_MODULE=ibus` as mentionned above, but also
2. `im-config` to rollback from `xim` to `ibus`

![Screenshot from 2023-05-30 10-06-50](https://github.com/dbeaver/dbeaver/assets/56821524/f184fefd-e4db-4ed2-9ed8-e4a96d7f781c)

My configuration:
- DBeaver Version 23.0.5.202305211610 (but same blinking bug with `xim` on previous version 22.3.3)
- Ubuntu 22.04.2 LTS
- GNOME 42.5
- X11
